### PR TITLE
feat: Edit curator assigned

### LINKED ISSUE #1981: Assign curator to curation
## 🎉 Description

For standard collections, the curation is created once changes are pushed but we'll need the entity created ahead of that event, so we can show the assignee. We need to call the curation POST once the curation gets assigned (that at that point is not created yet).

This PR covers the case to assign the curation to the users themself, show a confirmation modal and a toast in case showing the result of the action
  

* [ ] 🐛 This is a bug report.
* [X] 📈 This is a feature request.

### LINKED ISSUE #1989: Edit current assigned curator
## 🎉 Description

Once the curation is assigned, the assignee should be able to assign it to other curator. This is managed by a modal and a dropdown displaying a the full list of curators available. Show a toast once the operation is done.

* [ ] 🐛 This is a bug report.
* [X] 📈 This is a feature request.

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Closes #1981 
Closes #1989

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ module.exports = {
     '^.+\\.module\\.(css|sass|scss)$': 'identity-obj-proxy',
     '^!?raw-loader!.*$': 'identity-obj-proxy',
     '^three/.*$': 'identity-obj-proxy',
-    '^@babylonjs.*$': 'identity-obj-proxy'
+    '^@babylonjs.*$': 'identity-obj-proxy',
+    '^.*/modules/curations/collectionCuration/toasts.*$': 'identity-obj-proxy'
```
[the remark this answers is not in the document]
Comment (cazala) [reply jest.config.js:11]: would you leave a comment on the code explaining this?

File: src/components/Modals/EditCurationAssigneeModal/EditCurationAssigneeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const { onClose } = this.props
+    onClose()
+  }
+
+  render() {
+    const { name, metadata, isLoading, committeeMembers, collection, address } = this.props
+    const { type } = metadata
+    const { assignee } = this.state
+    const isReassigning = type === AssignModalOperationType.REASSIGN
+    console.log('assignee: ', assignee)
```
Comment (cazala) [inline src/components/Modals/EditCurationAssigneeModal/EditCurationAssigneeModal.tsx:37]: oops

File: src/components/Modals/EditCurationAssigneeModal/EditCurationAssigneeModal.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+              {isReassigning
+                ? t('curation_page.assign_modal.reassign.body')
+                : t('curation_page.assign_modal.self_assign.body', {
+                    emphasis: <b> {t('curation_page.assign_modal.self_assign.emphasis')} </b>
+                  })}
+            </div>
+            {isReassigning ? (
+              <SelectField
+                label={t('curation_page.curator')}
+                value={assignee ?? 'Pepe'}
```
Comment (cazala) [inline src/components/Modals/EditCurationAssigneeModal/EditCurationAssigneeModal.tsx:60]: :P

File: src/lib/api/builder.ts [hunk 4]
```diff
@@ -760,8 +760,8 @@ export class BuilderAPI extends BaseAPI {
     return fromRemoteCollectionCuration(curation)
   }
 
-  pushCuration(collectionId: string): Promise<void> {
-    return this.request('post', `/collections/${collectionId}/curation`)
+  pushCuration(collectionId: string, assignee?: string | null): Promise<void> {
+    return this.request('post', `/collections/${collectionId}/curation`, { curation: { assignee } })
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply src/lib/api/builder.ts:764]: Yeah, exactly! this is for creating a new one and the other is for updating an existing one

File: src/components/Modals/EditCurationAssigneeModal/EditCurationAssigneeModal.tsx (hunk 3 above)
Comment (juanmahidalgo) [reply src/components/Modals/EditCurationAssigneeModal/EditCurationAssigneeModal.tsx]: 😶‍🌫️

**Review by cazala — APPROVED** [review]

File: src/components/CurationPage/CollectionRow/CollectionRow.tsx [hunk 5]
```diff
@@ -105,17 +108,20 @@ export default class CollectionRow extends React.PureComponent<Props> {
...
+          <div className="edit-container">
             {curation?.assignee ? (
               <>
-                <Profile textOnly address={curation.assignee} />
-                {address === curation.assignee ? <> ({t('collection_row.you')})</> : null}
+                <div className="curator-name">
+                  <Profile textOnly address={curation.assignee} />
+                  {address === curation.assignee ? <> ({t('collection_row.you')})</> : null}{' '}
+                </div>
+                <Icon name="pencil" onClick={(e: any) => this.handleAssign(e, AssignModalOperationType.REASSIGN)} />
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply src/components/CurationPage/CollectionRow/CollectionRow.tsx]: oh yeah, forgot about that one! updated!

File: jest.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply jest.config.js]: for sure! done :)

**Pull request merged**